huaweicloud/services/vod/resource_huaweicloud_vod_transcoding_template_group.go

```diff
@@ -0,0 +1,592 @@
+	for i, v := range qualityInfo {
+		info := v.(map[string]interface{})
+		var outputFormat vod.QualityInfoFormat
+		switch info["output_format"].(string) {
+		case "MP4":
+			outputFormat = vod.GetQualityInfoFormatEnum().MP4
+		case "DASH":
+			outputFormat = vod.GetQualityInfoFormatEnum().DASH
+		case "DASH_HLS":
+			outputFormat = vod.GetQualityInfoFormatEnum().DASH_HLS
+		case "MP3":
+			outputFormat = vod.GetQualityInfoFormatEnum().MP3
+		case "ADTS":
+			outputFormat = vod.GetQualityInfoFormatEnum().ADTS
+		default:
+			outputFormat = vod.GetQualityInfoFormatEnum().HLS
+		}
+		qualityInfoOpts[i] = vod.QualityInfo{
+			Video:  buildVideoOpts(info["video"].([]interface{})),
+			Audio:  buildAudioOpts(info["audio"].([]interface{})),
+			Format: outputFormat,
+		}
+	}
+
+	return &qualityInfoOpts
+}
+
+func resourceTranscodingTemplateGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcVodV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating VOD client : %s", err)
+	}
+
+	createOpts := vod.TransTemplateGroup{
+		Name:                 d.Get("name").(string),
+		Type:                 vod.GetTransTemplateGroupTypeEnum().CUSTOM_TEMPLATE_GROUP,
+		Description:          utils.String(d.Get("description").(string)),
+		WatermarkTemplateIds: utils.ExpandToStringListPointer(d.Get("watermark_template_ids").([]interface{})),
+		Common:               buildCommonOpts(d.Get("common").(([]interface{}))),
+		QualityInfoList:      buildQualityInfoListOpts(d.Get("quality_info").(([]interface{}))),
+	}
+
+	if d.Get("auto_encrypt").(bool) {
+		createOpts.AutoEncrypt = utils.Int32(int32(1))
+	}
+
+	if d.Get("is_default").(bool) {
+		status := vod.GetTransTemplateGroupStatusEnum().E_1
+		createOpts.Status = &status
+	}
+
+	createReq := vod.CreateTemplateGroupRequest{
+		Body: &createOpts,
+	}
+	log.Printf("[DEBUG] Create VOD transcoding template group Options: %#v", createOpts)
+
+	resp, err := client.CreateTemplateGroup(&createReq)
+	if err != nil {
+		return diag.Errorf("error creating VOD transcoding template group: %s", err)
+	}
+
+	d.SetId(*resp.GroupId)
+
+	return resourceTranscodingTemplateGroupRead(ctx, d, meta)
+}
+
+func resourceTranscodingTemplateGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcVodV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating VOD client : %s", err)
+	}
+
+	resp, err := client.ListTemplateGroup(&vod.ListTemplateGroupRequest{GroupId: utils.String(d.Id())})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving VOD transcoding template group")
+	}
+
+	templateGroupList := *resp.TemplateGroupList
+
+	if len(templateGroupList) == 0 {
+		log.Printf("unable to retrieve VOD transcoding template group: %s", d.Id())
```

Comment:
tips: you can use diags to output the message in console with warning-level

Reply:
sure